src/plugins/http.js

```diff
@@ -43,7 +43,17 @@ function patch (http, methodName, tracer, config) {
 
       const req = request.call(this, options, callback)
 
-      req.on('socket', () => {
+      req.on('socket', socket => {
+        socket.once(socket.encrypted ? 'secureConnect' : 'connect', () => {
```

Comment:
If the socket never connects and is reused for other requests, won't this created a leak?